src/physics/utils/physics_data.F90

```diff
@@ -169,35 +172,52 @@ subroutine read_field_2d(file, std_name, var_names, timestep, buffer, mark_as_re
          mark_as_read_local = .true.
       end if
 
-      call cam_pio_find_var(file, var_names, found_name, vardesc, var_found)
-      if (.not. var_found) then
-          call cam_pio_find_var(file, [std_name], found_name, vardesc, var_found)
+      if (present(error_on_not_found)) then
+         error_on_not_found_local = error_on_not_found
+      else
+         error_on_not_found_local = .true.
       end if
 
-      if (var_found) then
+      var_found_local = .false.
+      call cam_pio_find_var(file, var_names, found_name, vardesc, var_found_local)
+      if (.not. var_found_local) then
+          call cam_pio_find_var(file, [std_name], found_name, vardesc, var_found_local)
+      end if
+
+      if (var_found_local) then
          if (masterproc) then
             write(iulog, *) 'Reading input field, ', trim(found_name)
             call shr_sys_flush(iulog)
          end if
-         call cam_read_field(found_name, file, buffer, var_found,             &
+         call cam_read_field(found_name, file, buffer, var_found_local,             &
               timelevel=timestep)
+
          if (mark_as_read_local) then
             call mark_as_read_from_file(std_name)
          end if
+
+         if (var_found_local) then
+            call shr_assert_in_domain(buffer, is_nan=.false.,                    &
+                 varname=trim(found_name),                                       &
+                 msg=subname//'NaN found in '//trim(found_name))
+         else
+            call endrun(subname//'Mismatch variable found in '//arr2str(var_names))
```

Comment:
This is a weird error check because, for what I can tell, the only way this condition would ever trigger is if there is a bug in the underlying PIO library, the input file has been corrupted somehow, or if the variable is removed from the file sometime between the `cam_pio_find_var` call and the subsequent  `cam_read_field` call.  So I might change this error message (and the one below) to be something like:

`'Unable to properly check the found variable "',trim(found_name),'" in the IC file.  Please double-check if the variable exists in the file, and that the file isn't corrupted or damaged.'`

How does that message sounds to folks?

Reply:
Sounds reasonable to me!